olive/evaluator/registry.py

```diff
@@ -0,0 +1,68 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import inspect
+import logging
+from typing import ClassVar, Dict
+
+logger = logging.getLogger(__name__)
+
+
+class Registry:
+    """Registry for olive model evaluators."""
+
+    _REGISTRY: ClassVar[Dict] = {}
+
+    @classmethod
+    def register(cls, name: str = None):
+        """Register an evaluator to the registry.
+
+        Args:
+            name (str): the name of the evaluator, if name is None, uses the class name
+
+        Returns:
+            Callable: the decorator function
+
+        """
+
+        def decorator(component):
+            component_name = name if name is not None else component.__name__
```

Comment:
nit: not important since users register their own evaluators but I recently made a PR to make all types case-insensitive by storing all keys in lowercase. https://github.com/microsoft/Olive/blob/64b79dcddd029ae2927637bb03074b46958763e2/olive/data/registry.py#L43

Reply:
Will make the change in the follow up PR.